projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/memberships.php

```diff
@@ -110,27 +121,39 @@ public function get_status_permission_check() {
 	/**
 	 * Do create a product based on data, or pass request to wpcom.
 	 *
-	 * @param object $request - request passed from WP.
+	 * @param WP_REST_Request $request - request passed from WP.
 	 *
 	 * @return array|WP_Error
 	 */
-	public function create_product( $request ) {
+	public function create_product( WP_REST_Request $request ) {
+		$is_editable             = isset( $request['is_editable'] ) ? (bool) $request['is_editable'] : null;
+		$type                    = isset( $request['type'] ) ? $request['type'] : null;
+		$buyer_can_change_amount = isset( $request['buyer_can_change_amount'] ) && (bool) $request['buyer_can_change_amount'];
+
```

Comment:
No need to change, we could just use the elvis operator here instead of ternary:
````
$is_editable         = isset( $request['is_editable'] ) ?: null;
$type                    = isset( $request['type'] ) ?: null;
````

Reply:
Another thing I've noticed is that we are building the same payload in the create_product and get_status methods. Maybe we could create a follow up task to not block this one in which we refactor mapping from the request to the payload to a separate method. I think it would also make the code more readable WDYT?

Reply:
>No need to change, we could just use the elvis operator here instead of ternary:

hmm... I had the impression that's not available in php 5.6, but 3v4l tells me otherwise, haha. For $type we could, but for $is_editable not because we typecast it to boolean (I prefer to have explicit typecast as much as possible, when it's feasible, ofc).

As for the payloads, I'm not really sure I follow. Looking at create_product and get_status, I don't see the same payload. 🤔  One it contains type and source while the other contains a lot more properties.

Reply:
You shouldn't need to cast to boolean since isset returns a boolean value.

Reply:
> As for the payloads, I'm not really sure I follow. Looking at create_product and get_status, I don't see the same payload. 🤔 One it contains type and source while the other contains a lot more properties.

Don't worry about it then, just ignore the comment.

Reply:
> Elvis operator is supported since 5.3. You shouldn't need to cast to boolean since isset returns a boolean value, therefore isset( $a ) ?: null will either return true or null.

Hmm... I always thought it was added later. 🤷  In which case this won't work for the `$type` variable then, because we expect the $type to be string. It doesn't matter in the end, it's just syntactic sugar.

> Don't worry about it then, just ignore the comment.

Sure :)